compat/mingw.c

```diff
@@ -3167,6 +3167,47 @@ int xwcstoutf(char *utf, const wchar_t *wcs, size_t utflen)
 	return -1;
 }
 
+#ifdef ENSURE_MSYSTEM_IS_SET
+static size_t append_system_bin_dirs(char *path, size_t size)
+{
+#if !defined(RUNTIME_PREFIX) || !defined(HAVE_WPGMPTR)
+	return 0;
+#else
+	char prefix[32768];
```

Comment:
would MAX_LONG_PATH be better here?

Reply:
Or, if this is a max env var limit rather than a disk path limit, some other suitably named constant ??

Reply:
> would MAX_LONG_PATH be better here?

I don't think so: I picked 32,768 because of the length restrictions of environment variables' values.